base/Base.jl

```diff
@@ -457,6 +457,14 @@ for match = _methods(+, (Int, Int), -1, get_world_counter())
     break   # only actually need to do this once
 end
 
+# triggers printing the report after a SIGINFO/SIGUSR1 profile request
+const PROF_PEEK_COND = Ref{Base.AsyncCondition}()
+function prof_peek()
+    wait(PROF_PEEK_COND[])
+    Base.require(Base.PkgId(Base.UUID("9abbd945-dff8-562f-b5e8-e1ebf5ef1b79"), "Profile")).profile_peek_report[]()
+    Threads.@spawn prof_peek()
```

Comment:
Why `@spawn` yourself? Using `while` seems more natural?

Reply:
Though the `wait` might be more likely to get blocked in this case, as it will always wait on the same thread..

Reply:
You don't need to re-spawn anything as long as the caller of `prof_peek` calls it in a `@spawn`. Typically, a `wait` call (and other yield points) enqueues the `current_task()` in the non-sticky queue before return. The scheduler can then migrate the task across worker threads when rescheduling this enqueued task.

OK, this is not technically the case *in general*. Tail-calling `@spawn` can make sense if you call `@async` or similar in the said task (as it will set the `stciky` flag of the parent). But presumably `Profile..profile_peek_report` does not call `@async`?

Reply:
Yeah we now have task-migration. so as long as non of the child functions creates an `@async` task all is good.

Reply:
Ok. Sounds good. Of course, all of this wouldn't be needed if we can call julia code directly from `signal-handling.c` but I believe that's not possible, right?

Reply:
No, I don't think we can call Julia in the signal handler or in the signal listener thread